app/src/main/java/com/duckduckgo/app/email/EmailJavascriptInterface.kt

```diff
@@ -17,35 +17,40 @@
 package com.duckduckgo.app.email
 
 import android.webkit.JavascriptInterface
-import timber.log.Timber
+import android.webkit.WebView
+import com.duckduckgo.app.browser.DuckDuckGoUrlDetector
+import com.duckduckgo.app.global.DispatcherProvider
+import kotlinx.coroutines.runBlocking
 
 class EmailJavascriptInterface(
     private val emailManager: EmailManager,
-    private val showNativeTooltip: () -> Unit
+    private val showNativeTooltip: () -> Unit,
+    private val webView: WebView,
+    private val urlDetector: DuckDuckGoUrlDetector,
+    private val dispatcherProvider: DispatcherProvider
 ) {
 
-    @JavascriptInterface
-    fun log(message: String) {
-        Timber.i("EmailInterface $message")
+    private fun isUrlFromDuckDuckGoEmail(): Boolean {
+        return runBlocking(dispatcherProvider.main()) {
+            val url = webView.url
+            (url != null && urlDetector.isDuckDuckGoEmailUrl(url))
```

Comment:
nit: I recently learned the kotlin idiomatic way to do this things and I kinda like it, don't need to change it but thought I would share it

```suggestion
            webView.url?.let { return urlDetector.isDuckDuckGoEmailUrl(it) }
            
            return false
```

Reply:
Noice, I like it :D